azure/scope/cluster.go

```diff
@@ -605,6 +605,17 @@ func (s *ClusterScope) SetFailureDomain(id string, spec clusterv1.FailureDomainS
 	s.AzureCluster.Status.FailureDomains[id] = spec
 }
 
+// FailureDomains returns the failure domains for the cluster.
+func (s *ClusterScope) FailureDomains() []string {
+	fds := make([]string, len(s.AzureCluster.Status.FailureDomains))
+	i := 0
```

Comment:
Total nit, but perhaps use
```go
	for i :=0; i<len(s.AzureCluster.Status.FailureDomains); i++ {
		fds[i] = s.AzureCluster.Stutus.FailureDomains[i]
	}
```

Reply:
I don't think we can do this since `s.AzureCluster.Status.FailureDomains` is a `map[string]FailureDomainSpec`.

Reply:
You are correct. I was thinking it was a slice. Thank you for the correction.